### PR TITLE
Add some fake components and tokens for demo and tests

### PR DESCRIPTION
**Pull request by pylapp**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

No issue

### Description

Add a fake `OUDSButton` with its component tokens so as to test typgoraphy, border and some view modifiers.
Improve also the public and internal API for view drawings.

### Motivation & Context

Have something more and new for demo, and go a bit further with components because of the lack of specifications!

### Types of change

- Refactoring (non-breaking change)

### Previews

None

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.github/CONTRIBUTING.md)

#### Accessibility

- (NA) My change follows accessibility good practices

#### Design

- (NA) My change respects the design guidelines of _Orange Unified Design System_

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.github/DEVELOP.md)
- [x]  I have added unit tests to cover my changes _(optional)_

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [x] The evolution have been tested and the project builds for iPhones and iPads
- [ ] Code review has been done by reviewers according to [CODEOWNERS file](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.github/CODEOWNERS)
- (NA) Design review has been done
- (NA) Accessibiltiy review has been done
- (NA) Q/A team has tested the evolution
- [x] Documentation has been updated if relevant
- [x] Internal files have been updated if relevant (like CONTRIBUTING, DEVELOP, THIRD_PARTY, CONTRIBUTORS, NOTICE)
- [x] CHANGELOG has been updated respecting [keep a changelog rules](https://keepachangelog.com/en/1.0.0/) and reference the issues

### CODE REVIEW

File: OUDS/Core/Components/Sources/Buttons/OUDSButton.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    // MARK: - Body
+
+    public var body: some View {
+        Button {
+            action()
+        } label: {
+            Text(text)
+                .systemFont(typography: theme.buttonTypography)
```
Comment (ludovic35) [inline OUDS/Core/Components/Sources/Buttons/OUDSButton.swift:61]: Can we add that in a ButtonStyle ?

File: OUDS/Core/Components/Sources/Buttons/OUDSButton.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                .padding(theme.buttonInternalSpacing)
+                .foregroundColor(colorScheme == .light
+                                 ? theme.buttonForegroundColorLight.color
+                                 : theme.buttonForegroundColorDark.color)
+                .modifier(BorderStyleModifier(theme.buttonBorderStyle,
+                                              theme.buttonBorderWidth,
+                                              theme.buttonBorderRadius,
+                                              theme.buttonBorderColorLight,
+                                              theme.buttonBorderColorDark))
+        }.frame(width: theme.buttonWidth, height: theme.buttonHeight)
```
Comment (ludovic35) [inline OUDS/Core/Components/Sources/Buttons/OUDSButton.swift:71]: What append on a long text and with dynamic font ?

File: OUDS/Core/Components/Sources/ViewModifiers/BorderStyleModifier.swift [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        self.colorLight = colorLight
+        self.colorDark = colorDark
+        if token != "solid" && token != "dashed" && token != "dotted" {
+            OUDSLogger.error("Unmanaged token: '\(token)'!")
+        }
+    }
+
+    // MARK: - Body
+
+    func body(content: Content) -> some View {
```
Comment (ludovic35) [inline OUDS/Core/Components/Sources/ViewModifiers/BorderStyleModifier.swift:66]: Add @ViewBuiilder and remove AnyView

File: OUDS/Core/Components/Sources/Buttons/OUDSButton.swift (hunk 1 above)
Comment (ludovic35) [reply OUDS/Core/Components/Sources/Buttons/OUDSButton.swift:61]: Can the Style be in the theme ?

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: OUDS/Core/Components/Sources/Buttons/OUDSButton.swift (hunk 1 above)
Comment (pylapp) [reply OUDS/Core/Components/Sources/Buttons/OUDSButton.swift:61]: Should be designed managed later, here are only dumb components for demo and tests. Indeed we may need to define and use such styles per themes.

File: OUDS/Core/Components/Sources/Buttons/OUDSButton.swift (hunk 2 above)
Comment (pylapp) [reply OUDS/Core/Components/Sources/Buttons/OUDSButton.swift:71]: Should be designed and managed later as we don't have specifications for components yet. These are dumb values.

**Pull request merged**